MobiusCore/Source/MobiusController.swift

```diff
@@ -21,54 +21,97 @@ import Foundation
 
 /// Defines a controller that can be used to start and stop MobiusLoops.
 ///
-/// If a loop is stopped and then started again, the new loop will continue from where the last
-/// one left off.
-public class MobiusController<Model, Event, Effect> {
+/// If a loop is stopped and then started again, the new loop will continue from where the last one left off.
+public final class MobiusController<Model, Event, Effect> {
     private let loopFactory: (Model) -> MobiusLoop<Model, Event, Effect>
-    private let lock = Lock()
 
-    private var viewConnectable: ConnectClosure<Model, Event>?
+    private var viewConnectable: AsyncDispatchQueueConnectable<Model, Event>?
     private var viewConnection: Connection<Model>?
     private var loop: MobiusLoop<Model, Event, Effect>?
     private var modelToStartFrom: Model
+    private let loopQueue: DispatchQueue
```

Comment:
I've been thinking about this. One of the requirements we have is that it needs to be possible to execute events and view model changes synchronously. That means that events need to be dispatchable on the main thread. Maybe that should even be the default. I'm thinking that the update function, which should be pure, is extremely unlikely to take significant amounts of time. There is a risk that model observers take time, and there is a risk that effect handling takes time. But probably, the best model is to always dispatch and process events on the main thread, and then make it easy to offload effect processing and model observation (if needed, less likely) to other threads.